### PR TITLE
Create Automatic Differentiation

### PR DESCRIPTION
**Pull request by StevenWaldrip**
See https://octave.discourse.group/t/automatic-differentiation/2078
First time trying to add a package so don't know what I'm doing.

### CODE REVIEW

File: packages/Automatic Differentiation [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+layout: "package"
+permalink: "https://github.com/StevenWaldrip/Automatic-Differentiation"
+description: >-
+  Automatic-Differentiation for Octave.
+icon: "https://raw.githubusercontent.com/gnu-octave/pkg-example/main/doc/icon.png"
```
Comment (mmuetzel) [inline packages/Automatic Differentiation:6]: If you have an icon that could represent your package, you could point to it here.
This is the icon that is shown for the package in the index.
Currently, that is this icon:
https://raw.githubusercontent.com/gnu-octave/pkg-example/main/doc/icon.png

File: packages/Automatic Differentiation [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+links:
+- icon: "far fa-copyright"
+  label: "GPL-3.0-or-later"
+  url: "https://github.com/StevenWaldrip/Automatic-Differentiation/blob/main/LICENSE"
```
Comment (mmuetzel) [inline packages/Automatic Differentiation:10]: Afaict, the file has been renamed to `COPYING`.

File: packages/Automatic Differentiation [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+- name: "Steven Waldrip"
+  contact: "steven.waldrip@gmail.com"
+versions:
+- id: "1"
+  date: "2022-07-15"
+  sha256: "bff441755f0d68596f2efd027fe637b5b6c52b722ffd6255bdb8a5f34ab4ef2a"
+  url: "https://github.com/gnu-octave/pkg-example/archive/1.1.0.tar.gz"
+  depends:
+  - "octave (>= 4.0.0)"
+  - "pkg"
```
Comment (mmuetzel) [inline packages/Automatic Differentiation:30]: Remove this block. This is just the example from the template.
IIUC, the top element in the `versions` section of this file is the one that will be displayed as the main one in the package index.

File: packages/Automatic Differentiation [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  date: "2020-09-02"
+  sha256: 
+  url: "https://github.com/StevenWaldrip/Automatic-Differentiation"
+  depends:
+  - "octave (>= 4.0.0)"
+  - "pkg"
+- id: "dev"
+  date:
+  sha256:
+  url: "https://github.com/StevenWaldrip/Automatic-Differentiation"
```
Comment (mmuetzel) [inline packages/Automatic Differentiation:41]: This should probably be https://github.com/StevenWaldrip/Automatic-Differentiation/archive/refs/heads/main.zip

File: packages/Automatic Differentiation [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+versions:
+- id: "1"
+  date: "2022-07-15"
+  sha256: "bff441755f0d68596f2efd027fe637b5b6c52b722ffd6255bdb8a5f34ab4ef2a"
+  url: "https://github.com/gnu-octave/pkg-example/archive/1.1.0.tar.gz"
+  depends:
+  - "octave (>= 4.0.0)"
+  - "pkg"
+- id: "1.0.0"
+  date: "2020-09-02"
```
Comment (mmuetzel) [inline packages/Automatic Differentiation:32]: Adjust the date for the first release.

**Review by mmuetzel — CHANGES_REQUESTED** [review]
Not a complete review. Just some remarks.

**Review by mmuetzel — APPROVED** [review]

**Comment by mmuetzel** [comment]
Looks good to me.

@siko1056: It looks like the CI was skipped. Do you know why?

**Comment by siko1056** [comment]
@mmuetzel The reason was that the package file was not matching the guidelines

```
packages/Automatic-Differentiation
packages/automatic-differentiation.yaml
```

Thus the CI pipeline ignored that change (for example changes I do on the website itself are also ignored).

**Comment by siko1056** [comment]
@StevenWaldrip congratulations for your package publication :tada: and @mmuetzel thank you for all you on-boarding guidance :slightly_smiling_face: :+1:

**Pull request merged**